### PR TITLE
feature: Value Entity Shopping Cart sample based on codegen

### PR DESCRIPTION
**Pull request by octonato**
Ports the shopping cart value entity sample to use the codegen. 

TODOs
- [x] fix the integration test (missing SDK library when running)
- [x] integrate in CI (similar to Java SDK and mvn projects)

### CODE REVIEW

**Comment by octonato** [comment]
I will defer the testing in CI because we need the a new version of `akkasls-scripts` and we are having issues with signing the artifacts. Once that is solved, we can make some work on CI integration.

**Review by ennru — APPROVED** [review]
LGTM.

**Comment by octonato** [comment]
This is rebased with `main` after merging the value entity counter PR #19. It's now failing because of permission issue. This time not during the integration tests, but during tests. 

Which is strange, because nothing changed in comparison to #19.

**Comment by pvlugter** [comment]
Removed the `sudo` in #25, which looks like it could help here.

File: samples/js-valueentity-shopping-cart/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,47 @@
-FROM node:14-buster-slim AS builder
+# This Dockerfile uses multi-stage build process.
+# See https://docs.docker.com/develop/develop-images/multistage-build/
+
+# Stage 1: Downloading dependencies and building the application
+FROM node:15.10.0-buster-slim AS builder
```
[the remark this answers is not in the document]
Comment (octonato) [reply samples/js-valueentity-shopping-cart/Dockerfile:5]: Strange, I don't remember to have touched that file. From what I can tell, this was from the template. Strange enough, if I start a new project from the template I get 14.17.

[the remark this answers is not in the document]
Comment (pvlugter) [reply samples/js-valueentity-shopping-cart/Dockerfile:5]: Ah, it was updated: https://github.com/lightbend/akkaserverless-npm-js/pull/53 That looks better. Thanks @andreaTP

[the remark this answers is not in the document]
Comment (octonato) [reply samples/js-valueentity-shopping-cart/Dockerfile]: I updated the node version. The two new samples we added were created when the template was using v15 (see https://github.com/lightbend/akkaserverless-npm-js/commit/629c5dce5ad4ba16288404ebb29fb5b37e6bbc1b#diff-189f46376e1ced30f8ddd9f3d7e095f2bf57ea32b1ae29487490c004448a0550). 

Now reverted to v14.17

**Review by octonato — COMMENTED** [review]
@ennru / @andreaTP, I think this is ready for a last review and merge. 

I will start on the event sourced one, based on top of this one.

**Review by ennru — APPROVED** [review]
Let's move this in.

**Pull request merged**